aea/protocols/default/message.py

```diff
@@ -59,20 +59,47 @@ def __init__(self, type: Optional[Type] = None,
         super().__init__(type=type, **kwargs)
         assert self.check_consistency(), "DefaultMessage initialization inconsistent."
 
+    @property
+    def type(self) -> Type:  # noqa: F821
+        """Get the type of the message."""
+        assert self.is_set("type"), "type is not set"
+        return DefaultMessage.Type(self.get("type"))
+
+    @property
+    def content(self) -> bytes:
+        """Get the content of the message."""
+        assert self.is_set("content"), "content is not set!"
+        return cast(bytes, self.get("content"))
+
+    @property
+    def error_code(self) -> ErrorCode:  # noqa: F821
+        """Get the error_code of the message."""
+        assert self.is_set("error_code"), "error_code is not set"
+        return DefaultMessage.ErrorCode(self.get("error_code"))
+
+    @property
+    def error_msg(self) -> str:
+        """Get the error message."""
+        assert self.is_set("error_msg"), "error_msg is not set"
+        return cast(str, self.get("error_msg"))
+
+    @property
+    def error_data(self) -> Dict[str, Any]:
+        """Get the data of the error message."""
+        assert self.is_set("error_data"), "error_msg is not set."
+        return cast(Dict, self.get("error_data"))
+
     def check_consistency(self) -> bool:
         """Check that the data is consistent."""
         try:
             ttype = DefaultMessage.Type(self.get("type"))
             if ttype == DefaultMessage.Type.BYTES:
-                assert self.is_set("content")
-                content = self.get("content")
-                assert isinstance(content, bytes)
+                assert isinstance(self.content, bytes), "Expect the content to be bytes"
```

Comment:
@marcofavorito and @Totoual  I am not sure the property should be used here. My worry is that in the property we cast to bytes, so this would always pass? I think we cannot use the properties here, instead we should recover from the body as done previously and then apply the checks.